imap/index.c

```diff
@@ -3391,6 +3392,9 @@ EXPORTED uint32_t index_finduid(struct index_state *state, uint32_t uid)
         else
             low = mid + 1;
     }
+
+    if (mode == FIND_EQ) return 0;
+    if (mode == FIND_GE) return high+1;
```

Comment:
Can this mode lead to an off-by-one overflow?

Reply:
The only place that FIND_GE is used is in nntpd, and its handled properly there.  If we returned state->exists rather than state->exists+1, nntpd would incorrectly access the last message if a UID greater than actually exists is requested.